cpp/src/io/csv/writer_impl.cu

```diff
@@ -268,8 +265,8 @@ struct column_to_strings_fn {
     // where modify() = duplicate the double quotes, if any; add 2bl quotes prefix/suffix;
     //}
     //
-    std::string delimiter{options_.get_inter_column_delimiter()};
-    predicate_special_chars pred{delimiter, stream_};
+    string_scalar delimiter{std::string{options_.get_inter_column_delimiter()}};
```

Comment:
As an additional benefit, the string scalar can now be constructed with a stream.